miden-tx/Cargo.toml

```diff
@@ -20,14 +20,16 @@ path = "tests/integration/main.rs"
 concurrent = ["miden-lib/concurrent", "miden-objects/concurrent", "miden-prover/concurrent", "std"]
 default = ["std"]
 std = ["miden-lib/std", "miden-objects/std", "miden-prover/std", "miden-verifier/std", "vm-processor/std"]
+testing = ["dep:miden-mock"]
```

Comment:
How would this affect publishing of the crate to crates.io? I think if we list `miden-mock` as a dependency, it needs to be published to crates.io as well (and maybe that's something we'll need to do anyway).

If we do want to publish `miden-mock` to crates.io - we should probably revisit it. For example, `miden-mock` may not be the right name any more (though, maybe it is fine). Also, we should probably check if it works correctly (e.g., the binary may be broken - and maybe it needs to be removed anyway, I'm not sure `no_std` support there works correctly etc.).

Reply:
> How would this affect publishing of the crate to crates.io? I think if we list miden-mock as a dependency, it needs to be published to crates.io as well (and maybe that's something we'll need to do anyway).

That is a good point, I didn't want to "force" the publishing of miden-mock, I just wanted to remove code duplication because the code started to diverge.

Reply:
Yep - understood. I'm not sure what is the right approach here. I don't think we can merge this PR as is as `miden-mock` is not in publishable state. But I'm also not sure if there is a way to reduce duplication without making `miden-mock` a dependency.

Reply:
Removed the commit that changes the mock. Please re-review.

I also would appreciate a decision on what to do with the mock. My current proposal is to delete the crate and move the utilities along side the structs they are related to. Maybe gated by `testing` and `std` if necessary.

Reply:
> My current proposal is to delete the crate and move the utilities along side the structs they are related to. Maybe gated by `testing` and `std` if necessary.

Yeah - that might be the best way forward. It's a pretty big task though. So, the question is when to do it. Let's create an issue for this (if one doesn't exist yet).